pkg/resources/cloudcontroller/vsphere.go

```diff
@@ -70,11 +70,18 @@ func vsphereDeploymentCreator(data *resources.TemplateData) reconciling.NamedDep
 				Labels: podLabels,
 			}
 
-			dep.Spec.Template.Spec.DNSPolicy, dep.Spec.Template.Spec.DNSConfig, err =
-				resources.UserClusterDNSPolicyAndConfig(data)
-			if err != nil {
-				return nil, err
+			if !data.IsKonnectivityEnabled() {
+				dep.Spec.Template.Spec.DNSPolicy, dep.Spec.Template.Spec.DNSConfig, err =
+					resources.UserClusterDNSPolicyAndConfig(data)
+				if err != nil {
+					return nil, err
+				}
+			} else {
+				// custom DNS resolver in not needed in Konnectivity setup
+				dep.Spec.Template.Spec.DNSPolicy = corev1.DNSClusterFirst
+				dep.Spec.Template.Spec.DNSConfig = nil
 			}
+
```

Comment:
that part of the code repeats quite often, could you make a function out of this? Maybe good place would be to create a new file under `pkg/resources/features.go` and have a place to check enabled feature gates? Or add this to `pkg/resources/resources.go` like we have for `UserClusterDNSPolicyAndConfig`?

Reply:
Or even could be a good idea to refactor `resources.UserClusterDNSPolicyAndConfig` and check inside if konnectivity is enabled? What do you think?

Reply:
good idea, let me try that